### PR TITLE
Initialize etc shadowdb and create homes

### PR DESCRIPTION
**Pull request by vcaputo**

### CODE REVIEW

File: dracut/99setup-root/initrd-setup-root [hunk 1]
```diff
@@ -18,6 +18,17 @@ do_setup_root() {
...
 
+    # This creates the modifiable users/groups in /sysroot/etc,
+    # initializing the shadow database in the process.
+    /usr/sbin/coreos-tmpfiles /sysroot
+
+    # Now that we have /sysroot/etc, we can use systemd-tmpfiles to create home,
+    # but it requires a chroot since --root is broken.  We also need to mount
+    # proc in the sysroot for the systemd-tmpfiles fchownat emulation.
+    mount -t proc none /sysroot/proc
+    chroot /sysroot systemd-tmpfiles --create baselayout-home.conf
```
Comment (marineam) [inline dracut/99setup-root/initrd-setup-root]: baselayout-home is only going to need `core` so why not just copy the user to the initrd? I could see a chroot being needed if you are handling arbitrary user provided configs but that's not the case here.

**Comment by vcaputo** [comment]
@marineam PTAL, thanks

**Comment by marineam** [comment]
LGTM

**Pull request merged**